carapace-server/src/main/java/org/carapaceproxy/EndpointMapper.java

```diff
@@ -53,11 +53,11 @@
 
     public abstract MapResult map(HttpRequest request, String userId, String sessionId, BackendHealthManager backendHealthManager, RequestHandler requestHandler);
 
-    public MapResult mapDefaultInternalError(HttpRequest request, String routeid) {
+    public MapResult mapInternalError(HttpRequest request, String routeid) {
```

Comment:
Maybe it is better to create a new class SimpleHTTPResponse that only contains the HTTP status code,  message, headers.....
This way it is clearer that the outcome cannot be *any* kind of action but only a static error code